chain/rust/src/genesis/shelley/raw.rs

```diff
@@ -0,0 +1,137 @@
+    #[serde(deserialize_with = "deserialize_string_from_number")]
+    pub decentralisationParam: String,
+    pub eMax: u64,
+    pub extraEntropy: ShelleyGenesisExtraEntropy,
+    pub keyDeposit: u64,
+    pub maxBlockBodySize: u64,
+    pub maxBlockHeaderSize: u64,
+    pub maxTxSize: u64,
+    pub minFeeA: u64,
+    pub minFeeB: u64,
+    pub minPoolCost: u64,
+    pub minUTxOValue: u64,
+    pub nOpt: u64,
+    pub poolDeposit: u64,
+    pub protocolVersion: ShelleyGenesisProtocolVersion,
+    // convert lossless JSON floats to string to avoid lossy Rust f64
+    #[serde(deserialize_with = "deserialize_string_from_number")]
+    pub rho: String,
+    // convert lossless JSON floats to string to avoid lossy Rust f64
+    #[serde(deserialize_with = "deserialize_string_from_number")]
+    pub tau: String,
+}
+
+#[allow(non_snake_case)]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ShelleyGenesisExtraEntropy {
+    pub tag: String,
+}
+
+#[allow(non_snake_case)]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ShelleyGenesisProtocolVersion {
+    pub major: u64,
+    pub minor: u64,
+}
+
+#[allow(non_snake_case)]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ShelleyGenesisStaking {
+    pub pools: HashMap<String, ShelleyGenesisPool>,
+    pub stake: HashMap<String, String>,
+}
+
+#[allow(non_snake_case)]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ShelleyGenesisPool {
+    pub cost: u64,
+    // convert lossless JSON floats to string to avoid lossy Rust f64
+    #[serde(deserialize_with = "deserialize_string_from_number")]
+    pub margin: String,
+    pub metadata: Option<ShelleyGenesisPoolMetadata>,
+    pub owners: Vec<String>,
+    pub pledge: u64,
+    pub publicKey: String,
+    pub relays: Vec<RelayTypeMap>,
+    pub rewardAccount: ShelleyGenesisRewardAccount,
+    pub vrf: String,
+}
+
+// TODO: there are other relay types, but I can't find the JSON type for them
+//       and I can't find any usage of them in the wild anyway
+// The key here defines the relay type
+// ex:
+// - single host address
+type RelayTypeMap = HashMap<String, ShelleyGenesisPoolSingleHotsRelay>;
+
+#[allow(non_snake_case)]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ShelleyGenesisPoolSingleHotsRelay {
+    pub IPv6: Option<String>,
+    pub port: Option<u16>,
+    pub IPv4: Option<String>
+}
+
+#[allow(non_snake_case)]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ShelleyGenesisPoolMetadata {
+    pub hash: String,
+    pub url: String,
+}
+#[allow(non_snake_case)]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ShelleyGenesisRewardAccount {
+    pub network: String,
+    pub credential: ShelleyGenesisCredential
+}
+#[allow(non_snake_case)]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ShelleyGenesisCredential {
+    // for some reason there actually is a space in the JSON key emitted by the Haskell node
+    #[serde(rename = "key hash")]
```

Comment:
It looks like cardano-node changed it so that both `key hash` and `keyHash` are accepted, so we have to handle both those possibilities here as well

Reply:
@SebastienGllmt You can use `#[serde(alias = "keyHash")]`

It can even be defined multiple times.

example:
```rust
#[derive(serde::Serialize, serde::Deserialize, Debug)]
pub struct Foo {
    #[serde(alias = "KeyHash")]
    #[serde(alias = "keyHash")]
    key_hash: String,
}

fn main() {
    let jsons = [
        "{ \"key_hash\": \"default\" }",
        "{ \"keyHash\": \"lowerCamelCase\" }",
        "{ \"KeyHash\": \"UpperCamelCase\" }",
        "{ \"keyhash\": \"should not parse\" }",
    ];
    for json in jsons {
        let foo: Result<Foo, _> = serde_json::from_str(json);
        println!("{json}      ->     {foo:?}");
    }
}
```

The first 3 are all accepted. First by it being the field name, and the next 2 for the `alias = ` marker.

Running:
```
{ "key_hash": "default" }      ->     Ok(Foo { key_hash: "default" })
{ "keyHash": "lowerCamelCase" }      ->     Ok(Foo { key_hash: "lowerCamelCase" })
{ "KeyHash": "UpperCamelCase" }      ->     Ok(Foo { key_hash: "UpperCamelCase" })
{ "keyhash": "should not parse" }      ->     Err(Error("missing field `key_hash`", line: 1, column: 33))
```